cekit/generator/base.py

```diff
@@ -21,10 +22,24 @@ class Generator(object):
     Args:
       descriptor_path - path to an image descriptor
       target - path to target directory
+      builder - builder type
       overrides - path to overrides file (can be None)
     """
 
-    def __init__(self, descriptor_path, target, overrides):
+    def __new__(cls, descriptor_path, target, builder, overrides):
+        if cls is Generator:
+            if 'docker' == builder or 'buildah' == builder:
+                from cekit.generator.docker import DockerGenerator as GeneratorImpl
+                logger.info('Generating files for Docker engine.')
```

Comment:
Do we want to mention the actual builder here? "Docker" may be misleading.

Reply:
good catch